fail2ban/server/action.py

```diff
@@ -394,20 +394,16 @@ def substituteRecursiveTags(cls, tags):
 					# recursive definitions are bad
 					#logSys.log(5, 'recursion fail tag: %s value: %s' % (tag, value) )
 					return False
-				elif found_tag in cls._escapedTags:
-					# Escaped so won't match
+				if found_tag in cls._escapedTags or not tags.has_key(found_tag):
+					# Escaped or missing tags - just continue on searching after end of match
+					# Missing tags are ok - cInfo can contain aInfo elements like <HOST> and valid shell
+					# constructs like <STDIN>.
+					m = t.search(value, m.end())
```

Comment:
>    and it's faster - start/reload of my f2b (with ca. 20 jails) improved up
>    to 0.5 second faster.

may be just by missing something it should've not missed? ;)  I think
such a speedup should still be in place even if you revert only this one
to the start+1